guide/creating-your-bot/command-handling.md

```diff
@@ -19,28 +19,30 @@ client.once('ready', () => {
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	const { commandName } = interaction;
-
-	if (commandName === 'ping') {
+	if (interaction.commandName === 'ping') {
 		await interaction.reply('Pong!');
-	} else if (commandName === 'beep') {
-		await interaction.reply('Boop!');
+	} else if (interaction.commandName === 'server') {
+		await interaction.reply(`Server name: ${interaction.guild.name}\nTotal members: ${interaction.guild.memberCount}`);
+	} else if (interaction.commandName === 'user') {
+		await interaction.reply(`Your tag: ${interaction.user.tag}\nYour id: ${interaction.user.id}`);
 	}
 });
 
 client.login(token);
 ```
 :::
 ::: code-group-item deploy-commands.js
-```sh:no-line-numbers
-npm install @discordjs/rest discord-api-types
-```
 ```js
+const { SlashCommandBuilder } = require('@discordjs/builders');
 const { REST } = require('@discordjs/rest');
 const { Routes } = require('discord-api-types/v9');
 const { clientId, guildId, token } = require('./config.json');
 
-const commands = [];
+const commands = [
```

Comment:
> the command handler is based on previous code

the previous code is from `creating-commands.md`, which has all three commands (ping, server, user) in the `deploy-commands.js` file. 

Not only that, but this code block is located at the top of the page and is simply there as a "these are the files from the previous section. we will be modifying these files to implement a command handler". The implementation of the command handler is then explained in the rest of the page. So i believe "replicating" the files from the previous section is appropriate here.

---
The current guide is divided into these sections:
- individual command files
  - move the commands (made from SlashCommandBuilder) from `deploy-commands.js` into its own file
  - move the reply of each command from `index.js` into its corresponding file
- reading command files
  - import the command files in `deploy-commands.js` using for loop
  - create a Collection in `index.js` and populate it using for loop
- dynamically executing commands
  - in the `index.js` file, use `.get()` and then `.execute()`

My refactor re-arranges them into these sections:
- implement command handler for the `deploy-command.js` file
  - move the commands (made from SlashCommandBuilder) into its own file
  - import the command files in `deploy-commands.js` using for loop
- implement command handler for the `index.js` file
  - move the reply of each command into its corresponding file
  - create a Collection in `index.js`, so we can dynamically execute commands using `.get()` and then `.execute()`
- summary
  - further explanation why it's a better approach to split up the commands into their own files.
  - by having the "structure" and the "execute" of a command in the same file, it becomes easier to develop the "execute" because you can see the "structure" at the same time.
    - for example, if you see that the structure has a USER and a STRING option, then you know you can `.getUser()` and `.getString()` in your execute.